### PR TITLE
[16.0][IMP] auth_oidc: Add AzureAD code flow provider

### PR DESCRIPTION
**Pull request by CRogos**
For an easier getting started I've updated the documentation and added pre-configured auth providers for Microsoft Azure AD

### CODE REVIEW

**Comment by sbidoul** [comment]
@adrienpeiffer could you have a quick look at this one?

File: auth_oidc/data/auth_oauth_data.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        <field name="css_class">fa fa-fw fa-windows</field>
+        <field name="body">Log in with Microsoft</field>
+    </record>
+    <record id="provider_azuread_single" model="auth.oauth.provider">
+        <field name="name">Azure AD Single Tenant</field>
+        <field name="flow">id_token_code</field>
+        <field name="token_map">upn:user_id upn:email</field>
+        <field
+            name="auth_endpoint"
+        >https://login.microsoftonline.com/{Tennant_id}/oauth2/v2.0/authorize</field>
```
Comment (adrienpeiffer) [inline auth_oidc/data/auth_oauth_data.xml]: Could you replace tennant_id by tenant_id ?

File: auth_oidc/data/auth_oauth_data.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <record id="provider_azuread_multi" model="auth.oauth.provider">
```
Comment (adrienpeiffer) [inline auth_oidc/data/auth_oauth_data.xml:3]: Should be enabled false by default

File: auth_oidc/readme/CONFIGURE.rst [hunk 3]
```diff
@@ -16,14 +14,24 @@ to the client, and in logs.
...
-* Validation URL: use the value of "OAuth2 token endpoint (v2)" from the Azure endpoints list
+.. image:: ..static/description/oauth-microsoft_azure-api_permissions.png
+
+.. image:: ..static/description/oauth-microsoft_azure-optional_claims.png
+
+* Provider Name: Azure AD Single Tenant
+* Client ID: Application (client) id
+* Client Secret: Client secret
 * Allowed: yes
+* replace {Tennant_id} in urls with your Azure tenant id
```
Comment (adrienpeiffer) [inline auth_oidc/readme/CONFIGURE.rst]: Could you replace tennant_id by tenant_id ?

File: auth_oidc/readme/CONFIGURE.rst [hunk 4]
```diff
@@ -16,14 +14,24 @@ to the client, and in logs.
...
-* Client ID: use the value of the OAuth2 autorization endoing (v2) from the Azure Endpoints list
-* Body: Azure SSO
-* Authentication URL: use the value of "OAuth2 autorization endpoint (v2)" from the Azure endpoints list
-* Scope: openid email
-* Validation URL: use the value of "OAuth2 token endpoint (v2)" from the Azure endpoints list
+.. image:: ..static/description/oauth-microsoft_azure-api_permissions.png
+
+.. image:: ..static/description/oauth-microsoft_azure-optional_claims.png
+
+* Provider Name: Azure AD Single Tenant
```
Comment (adrienpeiffer) [inline auth_oidc/readme/CONFIGURE.rst:25]: What's the differences between auth with single or multi ? pros/cons ?

**Review by adrienpeiffer — COMMENTED** [review]
Thanks for the contribution. Some questions

File: auth_oidc/readme/CONFIGURE.rst (hunk 4 above)
Comment (CRogos) [reply auth_oidc/readme/CONFIGURE.rst:25]: Is this sentence sufficient? Or should I add this link as well? 
https://learn.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps

Comment (adrienpeiffer) [reply auth_oidc/readme/CONFIGURE.rst:25]: it's fine

**Review by adrienpeiffer — APPROVED** [review]
Thanks for the contribution. It's clearer now.

@sbidoul This can be merged

**Comment by sbidoul** [comment]
Thanks!

/ocabot merge patch

**Pull request merged**